algorithms/vi-map-helpers/src/vi-map-landmark-quality-evaluation.cc

```diff
@@ -91,4 +144,175 @@ void resetLandmarkQualityToUnknown(
   }
 }
 
+void findTracksOfInferiorDuplicateLandmarkObservations(
+    const vi_map::VIMap& map, const vi_map::Landmark& landmark,
+    vi_map::TrackKeypointMap* tracks_with_keypoints) {
+  // Store landmark observations based on frame and store track ids
+  const vi_map::KeypointIdentifierList& landmark_observations =
+      landmark.getObservations();
+  std::unordered_map<
+      vi_map::VisualFrameIdentifier, vi_map::KeypointIdentifierList>
+      frames_with_observations;
+  vi_map::TrackKeypointMap track_ids_with_observations;
+  std::unordered_map<vi_map::KeypointIdentifier, int>
+      observations_with_track_id;
+  for (const auto& keypoint : landmark_observations) {
+    vi_map::KeypointIdentifierList& frame_observation =
+        frames_with_observations[keypoint.frame_id];
+    frame_observation.push_back(keypoint);
```

Comment:
I think the assignment by ref to a temporary is not necessary, just call `push_back` directly on it. And actually better use `emplace_back`.